### PR TITLE
Refactor Publisher App

### PR DESCRIPTION
**Pull request by steveoni**
This pull request update `admin/publisher/id` base on the refactored publisher app as specified in #79

### CODE REVIEW

**Comment by steveoni** [comment]
@risenW `metastore-lib` could not fetch the data for the `id` passed via `admin/publisher/id` , but it is working if the `id` is replaced with the repo name instead.
https://github.com/datopian/gift-portal/blob/f89c37a62c68dc4c24355bf4c6e403b515673d55/pages/admin/publisher/%5Bid%5D.js#L35

File: lib/Metadata.js [hunk 1]
```diff
@@ -70,4 +70,30 @@ export default class Metadata {
 
     return response
   }
+
+  /**
+   * 
+   * @param {Number} id 
+   * @param {Object} userInfo 
+   */
+  async fetchMetadata(id, userInfo, token) {
```
Comment (risenW) [inline lib/Metadata.js:79]: @steveoni This is a duplicate function. Just call the fetch method from Metastore: https://github.com/datopian/gift-portal/blob/e46611c02c686e46669bbc41cfd4662bc6feb96c/lib/Metastore.js#L38. It returns the metadata, including the id

File: pages/admin/publisher/[id].js [hunk 2]
```diff
@@ -1,22 +1,42 @@
...
+Publisher.getInitialProps = async (context) => {
+
+  const metadata = new Metadata()
+  const {req} = context
+  const session = await getSession({ req })
+  const { user } = session
+
+  const userToken = process.env.APP_GITHUB_KEY
+
+  const data = await metadata.fetchMetadata(context.query.id,user,userToken)
```
Comment (risenW) [inline pages/admin/publisher/[id].js:35]: So here, you pass the name of the repo to the fetch method. Remember name not ID. This returns the metadata plus the id which you can then use.

**Review by risenW — CHANGES_REQUESTED** [review]

File: pages/admin/publisher/[id].js (hunk 2 above)
Comment (steveoni) [reply pages/admin/publisher/[id].js:35]: `id` here is from `admin/publisher/id`

**Review by risenW — APPROVED** [review]

**Pull request merged**